includes/class-wc-regenerate-images.php

```diff
@@ -23,27 +23,88 @@ class WC_Regenerate_Images {
 	 */
 	protected static $background_process;
 
+	/**
+	 * Stores size being generated on the fly.
+	 *
+	 * @var string
+	 */
+	protected static $requested_size;
+
 	/**
 	 * Init function
 	 */
 	public static function init() {
-		include_once WC_ABSPATH . 'includes/class-wc-regenerate-images-request.php';
-		self::$background_process = new WC_Regenerate_Images_Request();
+		add_filter( 'wp_generate_attachment_metadata', array( __CLASS__, 'add_uncropped_metadata' ) );
 
-		if ( ! is_admin() ) {
-			// Handle on-the-fly image resizing.
+		// Resize WooCommerce images on the fly when browsing site through customizer as to showcase image setting changes in real time.
+		if ( is_customize_preview() ) {
 			add_filter( 'wp_get_attachment_image_src', array( __CLASS__, 'maybe_resize_image' ), 10, 4 );
 		}
 
-		if ( apply_filters( 'woocommerce_background_image_regeneration', true ) ) {
-			// Actions to handle image generation when settings change.
-			add_action( 'update_option_woocommerce_thumbnail_cropping', array( __CLASS__, 'maybe_regenerate_images_option_update' ), 10, 3 );
-			add_action( 'update_option_woocommerce_thumbnail_image_width', array( __CLASS__, 'maybe_regenerate_images_option_update' ), 10, 3 );
-			add_action( 'update_option_woocommerce_single_image_width', array( __CLASS__, 'maybe_regenerate_images_option_update' ), 10, 3 );
-			add_action( 'after_switch_theme', array( __CLASS__, 'maybe_regenerate_image_theme_switch' ) );
+		// Regenerate thumbnails in the background after settings change. Not ran on multisite to avoid multiple simultanious jobs..
+		if ( apply_filters( 'woocommerce_background_image_regeneration', ! is_multisite() ) ) {
+			include_once WC_ABSPATH . 'includes/class-wc-regenerate-images-request.php';
+
+			self::$background_process = new WC_Regenerate_Images_Request();
+
+			add_action( 'admin_init', array( __CLASS__, 'regenerating_notice' ) );
+			add_action( 'woocommerce_hide_regenerating_thumbnails_notice', array( __CLASS__, 'dismiss_regenerating_notice' ) );
+			add_action( 'customize_save_after', array( __CLASS__, 'maybe_regenerate_images' ) );
+			add_action( 'after_switch_theme', array( __CLASS__, 'maybe_regenerate_images' ) );
```

Comment:
I get notices when switching themes: `Deprecated: Non-static method WC_Regenerate_Images::maybe_regenerate_images() should not be called statically in /srv/www/woodev/wp-includes/class-wp-hook.php on line 288`.

Reply:
I do not have the class-wc-regenerate-images.php file in my wp-include folder :-(